packages/cxl-ui/src/components/cxl-credential.js

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { LitElement, html } from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+import QRCode from 'qrcode';
+import '@conversionxl/cxl-lumo-styles';
+import cxlCredentialStyles from '../styles/cxl-credential-css.js';
+
+@customElement('cxl-credential')
+export class CxlCredential extends LitElement {
+  static get styles() {
+    return [cxlCredentialStyles];
+  }
+
+  @property({ type: Number, attribute: 'credential-id' }) credentialId = 0;
+
+  @property({ type: String, attribute: 'full-name' }) fullName = '';
+
+  @property({ type: String }) title = '';
+
+  @property({ type: String }) type = '';
+
+  @property({ type: String }) date = '';
+
+  @property({ type: String, attribute: 'credential-url' }) credentialUrl = '';
```

Comment:
Rather than pass in the URL, could we generate the URL based off of the credential ID?

Reply:
That's already happening. The url attribute is optional. If empty, the element renders the value returned from a static function, adding the ID.

```
static get defaultUrl() {
    return 'https://cxl.com/institute/credential';
 }
```
Ideas on how to improve this? I'm not so convinced it's best, safest approach.

Reply:
My fault, I didn't see that in the `_renderQrCode` function. Perhaps use this?

```js
defaultUrl = 'https://cxl.com/institute/credential';

get url() {
  return `${this.credentialUrl || this.defaultUrl}/${this.credentialId}`;
}
```

It's not a huge improvement, though it is a few less lines of code.